<launch>
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video2" />
    <param name="brightness" value="0" />  
    <param name="contrast" value="34" />  
    <param name="saturation" value="32" />
    <param name="sharpness" value="28" />
    <!-- 摄像头HBV-V11(ov9281)格式只支持mmap -->
    <param name="io_method" value="mmap"/>
    <!-- 摄像头HBV-V11(ov9281)格式支持1280*720/800*600/640*480, 60/30/25HZ -->
    <param name="bStereo" value="true" /> 
    <param name="image_width" value="2560" />
    <param name="image_height" value="720" />
    <param name="framerate" value="60" />  
    <!-- 在摄像头频率固定输出时，调整次参数，降低频率(采样后,解码前)-->
    <param name="framerate_tb" value="0.0995" /> 
    <!-- 曝光时间延时(忽略) + 数据传输延时, 单位ns, (在摄像头HBV-V11 yuyv(不压缩)格式 1280 x 480 x 100 Hz条件下约98.1Hz(即传输速率为120545280字节=98.1*1280*480*2)) -->
    <param name="time_delay_per_byte" value="0" />
    <!-- 摄像头HBV-V11(ov9281)格式只支持 MJPG(压缩)或 YUYV(不压缩,等价于 yuy2) -->
    <param name="v4l_pixel_format" value="MJPG" /> 
    <!-- 输出格式: 需为sensor_msgs/image_encodings.h中之一, openvins代码默认使用mono8  -->
    <param name="publish_image_format" value="mono8" />     
    <param name="exposure_auto_priority" value="true" /> 
    <param name="exposure_absolute" value="20" /> 
    <param name="white_balance_temperature_auto" value="true" /> 
    <param name="white_balance_temperature" value="4600" /> 
    <param name="gain" value="32" /> 
    <param name="gamma" value="150" /> 
    <param name="hue" value="0" /> 
    <param name="backlight_compensation" value="72" /> 
    <param name="pan_absolute" value="0" /> 
    <param name="tilt_absolute" value="0" /> 
    <!-- 由于解码在imx8mq中时间较长,为提高帧率,接收后在另一个线程解码。TODO：但占用内存资源，后期需要把这部分代码优化及移植到算法代码程序里减少内存使用 -->
    <param name="process_image_in_other_thread" value="true" /> 
    <!-- 缓存数据的有效时间 -->
    <param name="image_valid_time" value="0.2" /> 
    <param name="camera_frame_id" value="ov9281" />
    <param name="camera_name" value="ov9281" />
    <!-- 请设置校正系数，只用于发布图像信息，发布的图像未校准 -->
    <!-- <param name="camera_info_url" value="file://$(find usb_cam)/launch/ov9281.yaml"/> -->

    <!-- Subscriber节点而不是读/dev/video, -->
    <param name="bSubscriberTopic" value="false" /> 
    <!-- 0:不发布mjpeg,发布mono8, 1:只发布mjpeg,不发布mono8, 2:发布mjpeg和mono8. 大于0时process_image_in_other_thread强转true -->
    <param name="bPUB_MJPEG" value="0" /> 
    <param name="MJPEG_Topic" value="/usb_cam/image_mjpeg" /> 

    <!-- 使能imx8 isp, dts配置需要对应mipi端口关闭isi并打开isp -->
    <param name="enable_imx_isp" value="false" /> 
    <!-- 使能imx8 dewarp, 用于校正鱼眼和广角镜头图像中引入的失真 -->
    <param name="enable_imx_dewarp" value="false" /> 
  </node>

  <!-- <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/usb_cam/image_raw"/>
    <param name="autosize" value="true" />
  </node> -->
</launch>
